docs/tools/testing/testing_icetea.md

```diff
@@ -0,0 +1,95 @@
+
+The full build process is:
+
+1. Build the nontest code (all code not under a `TESTS` folder), but do not link it. The resulting object files are placed in the build directory.
+1. Find all tests that match the given target and toolchain.
+1. Find all the C++ applications that are defined in the tests.
+1. For each discovered C++ application, build all of its source files and link it with the nontest code that was built in step 1.
+1. If specified, create a test specification file and place it in the given directory for use by testing tools. This is placed in the build directory by default when using Mbed CLI.
+
+#### Running tests
+
+You can run automated tests through Mbed CLI with the `--icetea` option. You can find more information about this in the [Mbed CLI test documentation] (/docs/tools/offline/cli-test-debug.md).
+
+The testing process requires tests to be built and that a test specification JSON file exist that describes these available tests. See the [test specification format](https://github.com/ARMmbed/greentea#test-specification-json-formatted-input).
+
+The test specification JSON is similar for the Greentea and Icetea tests.
+
+The Icetea tool handles the actual testing process. To read more about this tool, please visit its [GitHub repository](https://github.com/ARMmbed/icetea).
+
+#### Writing tests
+
+##### Writing test cases
+
+For writing your own Python test scripts, you can read further information from [Icetea Test Case API](https://github.com/ARMmbed/icetea/blob/master/doc/tc_api.md).
+
+##### Writing CLI applications
+
+To be able to run the commands given to the DUT, you need to provide a CLI application that runs on the DUT. There is a library that you can use to create your own CLI application; it is called [mbed-client-cli](https://github.com/ARMmbed/mbed-client-cli). You can also find example Mbed CLI applications from the `mbed-os` repository [test applications](https://github.com/ARMmbed/mbed-os/tree/master/TEST_APPS/device).
+
+### Debugging tests
+
+Debugging tests is a crucial part of the development and porting process. This section covers exporting the test, then driving the test with the test tools while the target is attached to a debugger.
+
+#### Exporting tests
+
+The most straightforward way to export a test is to copy the test application's source code from its test directory to your project's root. This way, the tools treat it like a normal application.
+
+You can find the path to the test application you wish to export by running the following command:
+
+```
+mbed test --compile-list -n <test name> --icetea
+```
+
+Once you've copied all of the test's source files to your project root, export your project:
+
+```
+mbed export -i <IDE name>
+```
+
+You can find your exported project in `projectfiles/<IDE>_<target>`.
```

Comment:
This isn't correct anymore. It should be removed from the greentea document as well, though that can be a separate change.

Reply:
Should this be "You can find your exported project files in the same folder you ran the export command" or totally removed?

Reply:
I would say "You can find your exported project in the root project directory." to match the patches I just pushed: https://github.com/ARMmbed/mbed-os-5-docs/pull/688